test/leafnode_test.go

```diff
@@ -1979,3 +2023,56 @@ func TestLeafNodeSendsRemoteSubsOnConnect(t *testing.T) {
 	_, leafExpect := setupConn(t, lc)
 	leafExpect(lsubRe)
 }
+
+func TestLeafNodeServiceImportLikeNGS(t *testing.T) {
+	server.SetGatewaysSolicitDelay(10 * time.Millisecond)
+	defer server.ResetGatewaysSolicitDelay()
+
+	ca := createClusterEx(t, true, "A", 3)
+	defer shutdownCluster(ca)
+	cb := createClusterEx(t, true, "B", 3, ca)
+	defer shutdownCluster(cb)
+
+	// Hang a responder off of cluster A.
+	opts := ca.opts[0]
+	url := fmt.Sprintf("nats://ngs:pass@%s:%d", opts.Host, opts.Port)
+	nc, err := nats.Connect(url)
+	if err != nil {
+		t.Fatalf("Error on connect: %v", err)
+	}
+	defer nc.Close()
+
+	// Create a queue subscriber to send results
+	nc.QueueSubscribe("ngs.usage.*", "ngs", func(m *nats.Msg) {
+		m.Respond([]byte("22"))
+	})
+
```

Comment:
Should you have `natsFlush(t, nc)` here? (to avoid flapping)

Reply:
yes/